repositories.bzl

```diff
@@ -37,115 +82,78 @@ IO_GRPC_GRPC_JAVA_OVERRIDE_TARGETS = {
         com_google_errorprone_error_prone_annotations()
-    if not omit_com_google_guava:
-        com_google_guava()
-    if not omit_com_google_guava_failureaccess:
+    if not native.existing_rule("com_google_guava_guava"):
+        com_google_guava_guava()
+    if not native.existing_rule("com_google_guava_failureaccess"):
         com_google_guava_failureaccess()
-    if not omit_com_google_j2objc_j2objc_annotations:
+    if not native.existing_rule("com_google_j2objc_j2objc_annotations"):
         com_google_j2objc_j2objc_annotations()
-    if not omit_com_google_protobuf:
-        com_google_protobuf()
-    if omit_com_google_protobuf_java:
-        fail("omit_com_google_protobuf_java is no longer supported and must be not be passed to grpc_java_repositories()")
-    if not omit_com_google_protobuf_javalite:
-        com_google_protobuf_javalite()
-    if not omit_com_google_truth_truth:
+    if not native.existing_rule("com_google_truth_truth"):
         com_google_truth_truth()
-    if not omit_com_squareup_okhttp:
-        com_squareup_okhttp()
-    if not omit_com_squareup_okio:
-        com_squareup_okio()
-    if not omit_io_grpc_grpc_proto:
-        io_grpc_grpc_proto()
-    if not omit_io_netty_buffer:
-        io_netty_buffer()
-    if not omit_io_netty_common:
-        io_netty_common()
-    if not omit_io_netty_transport:
-        io_netty_transport()
-    if not omit_io_netty_transport_native_epoll:
-        io_netty_transport_native_epoll()
-    if not omit_io_netty_codec:
-        io_netty_codec()
-    if not omit_io_netty_codec_socks:
-        io_netty_codec_socks()
-    if not omit_io_netty_codec_http:
-        io_netty_codec_http()
-    if not omit_io_netty_codec_http2:
-        io_netty_codec_http2()
-    if not omit_io_netty_handler:
-        io_netty_handler()
-    if not omit_io_netty_handler_proxy:
-        io_netty_handler_proxy()
-    if not omit_io_netty_resolver:
-        io_netty_resolver()
-    if not omit_io_netty_tcnative_boringssl_static:
-        io_netty_tcnative_boringssl_static()
-    if not omit_io_opencensus_api:
-        io_opencensus_api()
-    if not omit_io_opencensus_grpc_metrics:
-        io_opencensus_grpc_metrics()
-    if not omit_io_perfmark:
-        io_perfmark()
-    if not omit_javax_annotation:
-        javax_annotation()
-    if not omit_junit_junit:
+    if not native.existing_rule("com_squareup_okhttp_okhttp"):
+        com_squareup_okhttp_okhttp()
+    if not native.existing_rule("com_squareup_okio_okio"):
+        com_squareup_okio_okio()
+    if not native.existing_rule("io_netty_netty_buffer"):
+        io_netty_netty_buffer()
+    if not native.existing_rule("io_netty_netty_common"):
+        io_netty_netty_common()
+    if not native.existing_rule("io_netty_netty_transport"):
+        io_netty_netty_transport()
+    if not native.existing_rule("io_netty_netty_transport_native_epoll_linux_x86_64"):
```

Comment:
Not in dictionary order?

Reply:
I see now that codec is going backward. I had seen buffer, common, transport, transport_native_epoll and thought they were in order.

I didn't rearrange these from the previous version, but since the diff is already poor here but I'm also not doing anything special I'll go ahead and put the netty artifacts in sorted order. But I'm not going to rearrange them below; I don't want to create the diff noise and these will be deleted in a few months anyway.